packages/compose/src/hooks/use-media-query/index.js

```diff
@@ -6,12 +6,15 @@ import { useState, useEffect } from '@wordpress/element';
 /**
  * Runs a media query and returns its value when it changes.
  *
- * @param {string} query Media Query.
+ * @param {?string} query Media Query.
  * @return {boolean} return value of the media query.
  */
 export default function useMediaQuery( query ) {
 	const [ match, setMatch ] = useState( false );
 	useEffect( () => {
+		if ( ! query ) {
```

Comment:
I get that hooks need to be rendered deterministically, but it still feels odd to me that we'd want to support this use case of an empty `query` for `useMediaQuery`. Do the default hooks support this practice? i.e. `useCallback( undefined )` etc? Are there patterns in the hooks landscape for this sort of "forking" logic (i.e. do different behavior based on some condition)?

Reply:
Hi @aduth, I verified that useCallback( undefined ) returns undefined useCallback( false ) returns false, this behavior is equivalent to what we did useMediaQuery.
Without this pattern, I am not aware of any way we could implement this hook and avoid calling the media query listeners even when they are not needed because the value is simulated.